// portal-web/src/api/main.js
import request from '@/utils/request';

// 获取登录人信息
export function getMemUserInfo() {
  return request({
    url: '/member/memUser',
    method: 'get'
  });
}

// 获取分类列表
export function getCategoryList(params) {
  return request({
    url: '/product/proCategory/categoryList',
    method: 'get',
    params: params
  });
}
// 获取商品列表
export function getProductList(params) {
  return request({
    url: '/product/proProduct/page',
    method: 'get',
    params: params
  });
}
// 获取最新版本号
export function getLatestVersion(params) {
  return request({
    url: '/version/lastVersion',
    method: 'get',
    params: params
  });
}
// 添加配置信息
export function addSettingInfo(data) {
  return request({
    url: '/member/setting/add',
    method: 'post',
    data: data
  });
}
// 修改配置信息
export function updateSettingInfo(data) {
  return request({
    url: '/member/setting/update',
    method: 'post',
    data: data
  });
}
// 获取配置信息
export function getSettingInfo() {
  return request({
    url: '/member/setting/info',
    method: 'get'
  });
}
// 鼠币充值
export function rechargeRecord(data) {
  return request({
    url: '/member/memUserRechargeRecord',
    method: 'post',
    data: data
  });
}
// 鼠币充值
export function getRechargeRecord(id) {
  return request({
    url: '/member/memUserRechargeRecord/' + id,
    method: 'get'
  });
}
// 获取更多服务列表
export function getMoreServiceList(params) {
  return request({
    url: '/product/proProduct/moreServiceList',
    method: 'get',
    params: params
  });
}
// 获取收藏列表
export function getCollectionList(params) {
  return request({
    url: '/member/memUserCollection/list',
    method: 'get',
    params: params
  });
}
// 添加收藏
export function addCollection(data) {
  return request({
    url: '/member/memUserCollection',
    method: 'post',
    data: data
  });
}
// 删除收藏
export function deleteCollection(data) {
  return request({
    url: '/member/memUserCollection/delete',
    method: 'post',
    data: data
  });
}

// 获取多配置
export function getMultiConfig(params) {
  return request({
    url: '/config/multi',
    method: 'get',
    params
  });
}

// 获取单个配置
export function getSingleConfig(params) {
  return request({
    url: '/config/single',
    method: 'get',
    params
  });
}
