// portal-web/src/utils/cepio.js
import axios from 'axios';
import * as cepInterface from 'cep-interface';
// 注意：在 CEP 环境下，可以直接使用 Node.js 的原生模块
const fs2018 = window.require('fs');
const fs2022 = require('fs').promises;
const path = window.require('path');
/**
 * 下载远程文件并让 Photoshop 打开
 * @param {String} url - 远程图片地址
 * @param {String} fileName - 临时文件名 (例如 'temp_01.jpg')
 */
export const downloadAndOpen = async (url, folder, json) => {
  // cep 直接使用模块级导入，无需传参
  try {
    // 1. 获取系统临时文件夹路径 (一般是 AppData/Roaming 下的插件目录)
    const tempBase = cepInterface.getExtensionPath();
    const folderPath = path.join(tempBase, folder);
    const localPath = path.join(folderPath, getNameByFileUrl(url));
    // 2. 【关键步骤】检查并创建目录
    await createFolderIfNotExists(folderPath);
    // 3. 发起请求获取二进制流
    const response = await axios({
      url,
      method: 'GET',
      responseType: 'arraybuffer' // 获取原始二进制数据
    });
    // 4. 将数据转为 Buffer 并直接写入
    await createFile(localPath, response.data);
    // 5. 让 PS 打开
    // 关键：Windows 路径中的 \ 必须换成 /，否则 ExtendScript 会报错
    await openLocalFile(folder, localPath, json);
  } catch (error) {
    throw new Error(`下载流程出错: ${error.message}`);
  } finally {
    removeFolder(folder);
  }
};
export const downloadAndOpenBatch = async (files, folder, onProgress) => {
  const tempBase = cepInterface.getExtensionPath();
  const folderPath = path.join(tempBase, folder);

  await createFolderIfNotExists(folderPath);

  const total = files.length;
  const filePercents = new Array(total).fill(0);
  const localPaths = new Array(total).fill(null);

  const CONCURRENCY = 3; // 同时并发下载数

  const downloadFile = async (i) => {
    const file = files[i];
    const url = file.fileUrl;
    const fileName = file.sort + getNameByFileUrl(url);
    const localPath = path.join(folderPath, fileName);

    const response = await axios({
      url,
      method: 'GET',
      responseType: 'arraybuffer',
      onDownloadProgress: (progressEvent) => {
        const { loaded, total: fileTotal } = progressEvent;
        filePercents[i] = fileTotal ? Math.round((loaded / fileTotal) * 100) : 0;
        const overallPercent = Math.round(filePercents.reduce((sum, p) => sum + p, 0) / total);
        onProgress({ current: i + 1, total, fileName, filePercent: filePercents[i], overallPercent });
      }
    });

    await createFile(localPath, response.data);
    localPaths[i] = localPath;
  };

  // 1. 并发下载（限制同时 CONCURRENCY 个）
  const queue = [...files.keys()];
  await Promise.all(
    Array.from({ length: CONCURRENCY }, async () => {
      while (queue.length) {
        const i = queue.shift();
        try {
          await downloadFile(i);
        } catch (error) {
          throw new Error(`下载失败 [${files[i].fileUrl}]: ${error.message}`);
        }
      }
    })
  );

  // 2. 依次打开
  // try {
  // } catch (error) {
  //   throw new Error(`打开失败 [${localPaths}]: ${error.message}`);
  // }
  await openLocalFile(folder, localPaths[0], null);
};

// 创建目录（如果不存在）
async function createFolderIfNotExists(folderPath) {
  try {
    if (!fs2018.existsSync(folderPath)) {
      // recursive: true 确保即使多层目录不存在也能一次性创建
      await fs2018.mkdirSync(folderPath, { recursive: true });
    }
  } catch (error) {
    if (!fs2022.existsSync(folderPath)) {
      await fs2022.mkdirSync(folderPath, { recursive: true });
    }
  }
}
// 创建文件
async function createFile(filePath, data) {
  try {
    fs2018.writeFileSync(filePath, Buffer.from(data));
  } catch (error) {
    fs2022.writeFileSync(filePath, Buffer.from(data));
  }
}

/**
 * 打开本地的素材图片
 * @param {*} folder
 * @param {*} localPath
 * @returns
 */
export const openLocalFile = async (folder, localPath, json) => {
  const formattedPath = localPath.replace(/\\/g, '/');
  switch (folder) {
    case 'files':
      return cepInterface.evalExtendscript('createPsd("' + cepInterface.getExtensionPath() + '")');
    case 'caizhi':
      var newFolderAddress = formattedPath.slice(0, formattedPath.lastIndexOf('/'));
      return cepInterface.evalExtendscript('test2("' + newFolderAddress + '")');
    case 'sourceFile':
      return cepInterface.evalExtendscript('openSourceFile("' + formattedPath + '")');
    case 'localEmbedImageToMockup':
      // 将图片路径转换为对应的 PSD 路径传给 jsx
      var psdPath = formattedPath.replace(/\.[^.]+$/, '.psd');
      return cepInterface.evalExtendscript('embedImageToMockup("' + psdPath + '", null,"' + json + '")');
    default:
      return cepInterface.evalExtendscript('zong("' + encodeURIComponent(formattedPath) + '")');
  }
};

function getNameByFileUrl(fileUrl) {
  const queryIndex = fileUrl.indexOf('?');
  const slashIndex = fileUrl.lastIndexOf('/');

  const rawName = queryIndex > -1 ? fileUrl.substring(slashIndex + 1, queryIndex) : fileUrl.substring(slashIndex + 1);
  // 2. 将 URL 编码（如 %E6%B5%8B...）解码回中文
  try {
    return decodeURIComponent(rawName);
  } catch (e) {
    // 如果 URL 本身不是编码格式，报错时直接返回原样
    return rawName;
  }
}
// 增强版目录删除
export async function removeFolder(folder) {
  const folderPath = path.join(cepInterface.getExtensionPath(), folder);
  if (!fs2018.existsSync(folderPath)) return;
  try {
    const entries = await fs2022.readdir(folderPath, { withFileTypes: true });
    await Promise.all(
      entries.map(async (entry) => {
        const fullPath = path.join(folderPath, entry.name);
        entry.isDirectory() ? await removeFolder(fullPath) : await fs2022.unlink(fullPath);
      })
    );
    await fs2022.rmdir(folderPath);
  } catch (e) {
    await fs2018.readdirSync(folderPath).forEach((file) => {
      const curPath = path.join(folderPath, file);
      if (fs2018.lstatSync(curPath).isDirectory()) {
        // 递归使用
        removeFolder(curPath);
      } else {
        // 删除文件
        fs2018.unlinkSync(curPath);
      }
    });
    await fs2018.rmdirSync(folderPath);
  }
}
// 获取本地分类文件夹列表
export function getCommonLocalFolderList(dirPath) {
  try {
    // 读取目录内容
    return fs2022
      .readdir(dirPath, { withFileTypes: true })
      .filter((dirent) => dirent.isDirectory())
      .map((file) => {
        return { id: file.path, nameCn: file.name };
      });
  } catch (error) {
    return fs2018
      .readdirSync(dirPath)
      .filter((file) => {
        const fullPath = path.join(dirPath, file);
        return fs2018.statSync(fullPath).isDirectory();
      })
      .map((file) => {
        const curPath = path.join(dirPath, file);
        return { id: curPath, nameCn: file };
      });
  }
}

// 获取本地文件夹下的文件列表
export function getCommonLocalFileList(dirPath) {
  try {
    // 读取目录内容
    return fs2022
      .readdir(dirPath, { withFileTypes: true })
      .filter((dirent) => isImageFile(dirent))
      .map((file) => {
        return { thumbnail: file.path, nameCn: file.name, id: file.name };
      });
  } catch (error) {
    return fs2018
      .readdirSync(dirPath)
      .filter((file) => {
        const fullPath = path.join(dirPath, file);
        return isImageFile(fullPath);
      })
      .map((file) => {
        const curPath = path.join(dirPath, file);
        return { thumbnail: curPath, nameCn: file, id: file };
      });
  }
}
/**
 * 判断文件是否是图片
 * @param {string} fullPath - 文件路径
 * @returns {boolean} 是否是图片文件
 */
function isImageFile(fullPath) {
  const imageExtensions = ['.jpg', '.jpeg', '.png', '.gif', '.bmp', '.tif', '.webp', '.svg', '.ico'];
  try {
    const file = fs2018.statSync(fullPath);
    // 排除目录
    if (file.isDirectory()) return false;
    // 排除0字节文件
    if (file.size === 0) return false;
    // 获取扩展名并检查
    const ext = path.extname(fullPath).toLowerCase();
    return imageExtensions.includes(ext);
  } catch (error) {
    console.error(`无法检查文件: ${fullPath}`, error);
    return false;
  }
}
