// @vuepress/core/.temp/internal/routes.js
/**
 * Generated by "@vuepress/internal-routes"
 */

import { injectComponentOption, ensureAsyncComponentsLoaded } from '@app/util'
import rootMixins from '@internal/root-mixins'
import GlobalLayout from "D:\\soft\\JetBrains\\workspace\\tenant-plugin\\portal-web\\docs\\node_modules\\@vuepress\\core\\lib\\client\\components\\GlobalLayout.vue"

injectComponentOption(GlobalLayout, 'mixins', rootMixins)
export const routes = [
  {
    name: "v-51f41a65",
    path: "/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-51f41a65").then(next)
    },
  },
  {
    path: "/index.html",
    redirect: "/"
  },
  {
    name: "v-58a3e786",
    path: "/account/member.html",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-58a3e786").then(next)
    },
  },
  {
    name: "v-5997c918",
    path: "/SCREENSHOT-GUIDE.html",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-5997c918").then(next)
    },
  },
  {
    name: "v-425be463",
    path: "/account/recharge.html",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-425be463").then(next)
    },
  },
  {
    name: "v-31c3dd6e",
    path: "/account/setting.html",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-31c3dd6e").then(next)
    },
  },
  {
    name: "v-3c2a7e78",
    path: "/guide/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-3c2a7e78").then(next)
    },
  },
  {
    path: "/guide/index.html",
    redirect: "/guide/"
  },
  {
    name: "v-037370ae",
    path: "/guide/import-file.html",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-037370ae").then(next)
    },
  },
  {
    name: "v-cb1cd206",
    path: "/guide/local-material.html",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-cb1cd206").then(next)
    },
  },
  {
    name: "v-383970ee",
    path: "/guide/login.html",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-383970ee").then(next)
    },
  },
  {
    name: "v-2284f841",
    path: "/guide/material.html",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-2284f841").then(next)
    },
  },
  {
    name: "v-17017115",
    path: "/guide/more-service.html",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-17017115").then(next)
    },
  },
  {
    name: "v-234f4fff",
    path: "/guide/psd-file.html",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-234f4fff").then(next)
    },
  },
  {
    name: "v-d422c52e",
    path: "/guide/source-file.html",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-d422c52e").then(next)
    },
  },
  {
    name: "v-9bdf5c6e",
    path: "/guide/tools.html",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-9bdf5c6e").then(next)
    },
  },
  {
    path: '*',
    component: GlobalLayout
  }
]
